lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1622,6 +1624,93 @@ def get_configured_disk_size(self, expected_disk_spec):
         self.module.fail_json(
             msg="No size, size_kb, size_mb, size_gb or size_tb attribute found into disk configuration")
 
+    def vmdk_disk_path_split(self, vmdk_path):
+        """
+        Takes a string in the format
+
+            [datastore_name] /path/to/vm_name.vmdk
+
+        Returns a tuple with multiple strings:
+
+        1. datastore_name: The name of the datastore (without brackets)
+        2. vmdk_fullpath: The "/path/to/vm_name.vmdk" portion
+        3. vmdk_filename: The "vm_name.vmdk" portion of the string (os.path.basename equivalent)
+        4. vmdk_folder: The "/path/to/" portion of the string (os.path.dirname equivalent)
+        """
+        try:
+            datastore_name = re.match(r'^\[(.*?)\]', vmdk_path, re.DOTALL).groups()[0]
+            vmdk_fullpath = re.match(r'\[.*?\] (.*)$', vmdk_path).groups()[0]
+            vmdk_filename = os.path.basename(vmdk_fullpath)
+            vmdk_folder = os.path.dirname(vmdk_fullpath)
+        except (IndexError, AttributeError) as e:
+            raise RuntimeError("Bad path for filename disk vmdk image")
+
+        return (datastore_name, vmdk_fullpath, vmdk_filename, vmdk_folder)
+
+    def find_vmdk(self, vmdk_path):
```

Comment:
To add this logic to vmware_guest_disk, this method will move into module_utils/vmware.py.